pulp_ansible/tests/unit/test_search.py

```diff
@@ -0,0 +1,157 @@
+import hashlib
+import itertools
+import os
+import random
+import string
+
+from django.core.files.uploadedfile import SimpleUploadedFile
+from django.db import connection
+from django.test import TestCase
+
+from pulp_ansible.app.models import (
+    AnsibleDistribution,
+    AnsibleRepository,
+    Collection,
+    CollectionVersion,
+    Tag,
+)
+from pulp_ansible.tests.unit.helpers import make_cv_tarball
+from pulpcore.plugin.models import Artifact, ContentArtifact
+
+
+class TestSearchUtil(TestCase):
+
+    repo = None
+    collections = None
+
+    def setUp(self):
+
+        # make a couple repositories to distrbute the collections around
+        #   this is also to prepare for future cross-repo searching
+        self.repos = {}
+        for repo_name in ["published1", "published2"]:
+            self.repos[repo_name] = AnsibleRepository(name=repo_name)
+            self.repos[repo_name].save()
+
+        # make a distribution for each repo?
+        for repo_name, repo in self.repos.items():
+            AnsibleDistribution.objects.create(name=repo_name, base_path=repo_name, repository=repo)
+
+        # make some tarballs
+        self.collections = {}
+        for i in range(0, 2):
+            namespace_name = "".join([random.choice(string.ascii_lowercase) for x in range(0, 5)])
+            for j in range(0, 3):
+                collection_name = "".join(
+                    [random.choice(string.ascii_lowercase) for x in range(0, 5)]
+                )
+                for v in range(1, 3):
+                    vstring = f"1.0.{v}"
+                    tarfn = make_cv_tarball(namespace_name, collection_name, vstring)
+                    self.collections[(namespace_name, collection_name, vstring)] = {
+                        "tags": ["tag" + namespace_name + collection_name + vstring],
+                        "tar": tarfn,
+                    }
+
+        # "import" the collections
+        specs = sorted(list(self.collections.keys()))
+        for ids, spec in enumerate(specs):
+            cdata = self.collections[spec]
+
+            this_repo_name = random.choice(list(self.repos.keys()))
+            this_repo = self.repos[this_repo_name]
+            self.collections[spec]["repo"] = this_repo
+            self.collections[spec]["repo_name"] = this_repo_name
+
+            # make an artifact
+            rawbin = open(cdata["tar"], "rb").read()
+            artifact = Artifact.objects.create(
+                sha224=hashlib.sha224(rawbin).hexdigest(),
+                sha256=hashlib.sha256(rawbin).hexdigest(),
+                sha384=hashlib.sha384(rawbin).hexdigest(),
+                sha512=hashlib.sha512(rawbin).hexdigest(),
+                size=os.path.getsize(tarfn),
+                file=SimpleUploadedFile(tarfn, rawbin),
+            )
+            artifact.save()
```

Comment:
Since we are only checking the validity of the search vector field on the CollectionVersion I don't think the Artifact needs to be created.

Reply:
Yep, I'll try taking this out.